core/src/main/java/org/apache/iceberg/util/ParallelIterable.java

```diff
@@ -98,6 +99,22 @@ private boolean checkTasks() {
 
       for (int i = 0; i < taskFutures.length; i += 1) {
         if (taskFutures[i] == null || taskFutures[i].isDone()) {
+          if (taskFutures[i] != null) {
+            // check for task failure and re-throw any exception
+            try {
+              taskFutures[i].get();
+            } catch (ExecutionException e) {
+              if (e.getCause() instanceof RuntimeException) {
+                // rethrow a runtime exception
+                throw (RuntimeException) e.getCause();
+              } else {
+                throw new RuntimeException("Failed while running parallel task", e.getCause());
+              }
+            } catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)